src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/DSL.kt

```diff
@@ -80,11 +87,17 @@ class TestData(private val repository: Repository, private val courseAllocationB
     active: Boolean,
     dsl: ProgramServiceDsl.() -> Unit,
   ): ProgramService =
-    repository.save(ProgramServiceBuilder(repository, programCode, programId, description, active).apply(dsl))
+    programServiceBuilderFactory!!.builder()
+      .let { builder ->
+        builder.build(programCode, programId, description, active)
+          .also {
+            builder.apply(dsl)
+          }
+      }
```

Comment:
The entity graph of ProgramService -> CourseActivity -> [CourseSchedule, CourseScheduleRule, CourseActivityPayRate] stands alone from OffenderBooking and provides an example of how I think the DSL should work.

So at the top level `programService` can be added to `nomisDataBuidler.build {`:
* we get a builder from the factory, call `builder.build(...)` which creates the instances and saves it
* we then call `builder.apply(dsl)` to continue building down the entity chain
* the ProgramService returned from calling `build` is returned out of the function so it can be assigned to a variable in the test